File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
from heapq import heapify, heappop, heappush

def solution(scoville, K):
    answer = 0
    
    heapify(scoville)
    while len(scoville)>1:
        if scoville[0] < K:
            heappush(scoville, heappop(scoville)+heappop(scoville)*2)
            answer += 1
        else:
            break
    
    if scoville[0] < K:
        answer = -1
    
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        int answer = 0;
        
        for (int x : scoville) 
            heap.offer(x);

        while (heap.size() > 1 && heap.peek() < K) {
            heap.offer(heap.poll() + heap.poll() * 2);
            answer ++;
            continue;
        }
        
        return heap.poll() < K ? -1 : answer;
    }
}